internal/security/resource_service.go

```diff
@@ -152,39 +159,45 @@ func (s securityResourceService) DeleteWhitelist(ctx context.Context, cluster Cl
 	logCtx := map[string]interface{}{"clusterID": cluster.GetID(), "whiteListItem": whitelistItemName}
 	s.logger.Info("creating whitelist item ...", logCtx)
 
-	wlClient, err := s.getWhiteListsClient(ctx, cluster)
+	cli, err := s.getClusterClient(ctx, cluster)
 	if err != nil {
 		return err
 	}
 
-	if err := wlClient.Delete(whitelistItemName, metav1.NewDeleteOptions(0)); err != nil {
+	whiteListItem, err := s.getWhitelist(ctx, cluster, whitelistItemName)
+	if err != nil {
+		return err
+	}
 
+	if err := cli.Delete(ctx, whiteListItem); err != nil {
 		return errors.WrapIf(err, "failed to delete whitelist")
 	}
 
 	s.logger.Info("whitelist item successfully deleted", logCtx)
 	return nil
 }
 
-func (s securityResourceService) getWhiteListsClient(ctx context.Context, cluster Cluster) (securityClientV1Alpha.WhiteListInterface, error) {
-	securityClientSet, err := s.getClusterClientSet(ctx, cluster)
+func (s securityResourceService) getWhitelist(ctx context.Context, cluster Cluster, whitelistItemName string) (*securityV1Alpha.WhiteListItem, error) {
+	logCtx := map[string]interface{}{"clusterID": cluster.GetID()}
+	s.logger.Info("get whitelist item ...", logCtx)
+
+	cli, err := s.getClusterClient(ctx, cluster)
 	if err != nil {
-		return nil, errors.WrapIf(err, "failed to create security config")
+		return nil, errors.WrapIf(err, "failed to get audit client")
 	}
 
-	return securityClientSet.Whitelists(), nil
-}
-
-func (s securityResourceService) getAuditClient(ctx context.Context, cluster Cluster) (securityClientV1Alpha.AuditInterface, error) {
-	securityClientSet, err := s.getClusterClientSet(ctx, cluster)
-	if err != nil {
-		return nil, errors.WrapIf(err, "failed to create security config")
+	whilteListItem := &securityV1Alpha.WhiteListItem{}
```

Comment:
```suggestion
	whiteListItem := &securityV1Alpha.WhiteListItem{}
```

Reply:
Nice catch, I used it consistent, so I'll have to fix all of them.